src/esp/sensor/PinholeCamera.cpp

```diff
@@ -28,8 +29,50 @@ void PinholeCamera::setProjectionParameters(SensorSpec::ptr spec) {
   hfov_ = std::atof(spec_->parameters.at("hfov").c_str());
 }
 
-void PinholeCamera::setProjectionMatrix(gfx::RenderCamera& targetCamera) {
+PinholeCamera& PinholeCamera::setProjectionMatrix(
+    gfx::RenderCamera& targetCamera) {
   targetCamera.setProjectionMatrix(width_, height_, near_, far_, hfov_);
+  return *this;
+}
+
+PinholeCamera& PinholeCamera::setTransformationMatrix(
+    gfx::RenderCamera& targetCamera) {
+  CORRADE_ASSERT(!scene::SceneGraph::isRootNode(targetCamera.node()),
+                 "PinholeCamera::setTransformationMatrix: target camera cannot "
+                 "be on the root node of the scene graph",
+                 *this);
+  Magnum::Matrix4 absTransform = this->node().absoluteTransformation();
```

Comment:
@bigbike explained this logic to me offline but it wasn't initially obvious to me what was going on here. It all just seem like a no-op. But @bigbike explained that we are correcting for the fact that numerical errors in the rotation basis don't apply evenly so we need to correct for this by regenerating the basis.

Would be nice to have a comment to this affect with perhaps a reference citation. Its always nice when fancy math comes with a citation.